src/main/scala/com/ing/wbaa/airlock/proxy/provider/LineageProviderAtlas.scala

```diff
@@ -1,147 +1,78 @@
-  }
+    def readOrWriteLineage(method: String, bucket: String): Future[LineagePostGuidResponse] = {
+      logger.debug(s"Creating $method lineage for request to ${lh.method.value} file ${lh.bucketObject} at $bucket at $timestamp")
+      postEnities(userName, host, bucket, lh.bucketObject, method, lh.contentType, lh.clientType, timestamp)
+    }
 
-  // file entity delete
-  // for now it is just deleting file entity and no related objects like eg. aws_cli_script, which uploaded or downloaded
-  // file to bucket. Once we delete aws_cli_script object we will lose track of whats has been deleted
-  // We need to come up with process of tracking file delete
-  private def deleteEntities(typeName: String, entityName: String): Future[LineagePostGuidResponse] =
-    for {
-      entityGuid <- getEntityGUID(typeName, entityName)
-      _ <- deleteEntity(entityGuid)
+    def delLineage(method: String): Future[LineageGuidResponse] = {
+      logger.debug(s"Creating Delete lineage for request to ${lh.method} file ${lh.bucketObject} at ${lh.bucket} at $timestamp")
+      deleteEntities("DataFile", lh.bucketObject)
+    }
 
-    } yield LineagePostGuidResponse("", "", entityGuid, "")
+    // we only report lineage for object operations. We do not track bucket create / delete etc.
+    if (lh.bucket.length > 1 && lh.bucketObject != "emptyObject") {
+      lh.method match {
+        // get object
+        case HttpMethods.GET if lh.queryParams.isEmpty || lh.queryParams.contains("encoding-type") =>
+          readOrWriteLineage("read", lh.bucket)
 
-  def createLineageFromRequest(httpRequest: HttpRequest, userSTS: User): Future[LineagePostGuidResponse] = {
+        // put object
+        case HttpMethods.PUT if lh.queryParams.isEmpty && lh.copySource.isEmpty => readOrWriteLineage("write", lh.bucket)
 
-    val timestamp = System.currentTimeMillis()
-    // createLineageFromRequest is used in RequestHandlerS3 where there is no notion of S3Request. It seems easier to
-    // repeat httpRequest parsing here for needed values
-    val host = httpRequest.uri.authority.host.address()
-    val path = httpRequest.uri.path
-    val bucket = path.toString.split("/").toList.lift(1).getOrElse("notDef")
-    val bucketObject = s"${path.toString.split("/").toList.lift(2).getOrElse("emptyObject")}"
-    val method = httpRequest.method
-    val contentType = httpRequest.entity.contentType
-    val userName = userSTS.userName.value
+        // put object - copy
+        // if contains header x-amz-copy-source
+        case HttpMethods.PUT if lh.copySource.getOrElse("").length > 0 =>
+          readOrWriteLineage("read", lh.copySource.get)
+          readOrWriteLineage("write", lh.bucket)
+
+        // post object (complete multipart)
+        // aws request eg. POST /ObjectName?uploadId=UploadId and content-type application/xml
+        case HttpMethods.POST if lh.queryParams.getOrElse("").contains("uploadId") => readOrWriteLineage("write", lh.bucket)
```

Comment:
Without .getOrElse("") it does the same
```suggestion
        case HttpMethods.POST if lh.queryParams.contains("uploadId") => readOrWriteLineage("write", lh.bucket)
```